src/sentry/integrations/slack/event_endpoint.py

```diff
@@ -77,42 +105,55 @@ def unfurl_incidents(integration, incident_map, event_id_by_url=None):
         return {}
 
     return {
-        v: build_incident_attachment(
+        link.url: build_incident_attachment(
             action=None,
-            incident=results[k],
+            incident=results[link.args["incident_id"]],
         )
-        for k, v in incident_map.items()
-        if k in results
+        for link in links
+        if link.args["incident_id"] in results
     }
 
 
+# XXX: The regex matchers could be more tightly bound to our configured domain,
+# but slack limits what we can unfurl anyways so its probably safe
+_link_handlers = {
+    LinkType.ISSUES: Handler(
+        fn=unfurl_issues,
+        matcher=re.compile(
+            r"^https?\://[^/]+/[^/]+/[^/]+/issues/(?P<issue_id>\d+)(?:/events/(?P<event_id>\w+))?"
+        ),
+        arg_types={
+            "issue_id": int,
+            "event_id": str,
+        },
+    ),
+    LinkType.INCIDENTS: Handler(
+        fn=unfurl_incidents,
+        matcher=re.compile(
+            r"^https?\://[^/]+/organizations/(?P<org_slug>[^/]+)/incidents/(?P<incident_id>\d+)"
```

Comment:
Mind fixing this while you're here? The url is https://sentry.io/organizations/sentry/alerts/rules/details/32/, so `r"^https?\://[^/]+/organizations/(?P<org_slug>[^/]+)/alerts/rules/details/(?P<incident_id>\d+)"`